File: src/pay_digitalwallets_sdk_ios/ios/Classes/Data/ErrorCodes.swift
// Copyright © 2025 Olo Inc. All rights reserved.
// This software is made available under the MIT License (See LICENSE file)
//
//  ErrorCodes.swift
//  Pods
//
//  Created by Richard Dowdy on 11/19/25.
//

import Foundation

public class ErrorCodes {
    // Method Call Rejection Codes
    public static let InvalidParameter = "InvalidParameter"
    public static let MissingParameter = "MissingParameter"
    public static let UnexpectedError = "UnexpectedError"
    public static let UnexpectedParameterType = "UnexpectedParameterType"
    public static let ApplePayUnsupported = "ApplePayUnsupported"
    public static let SdkNotConfigured = "SdkNotConfigured"
    public static let EmptyMerchantId = "EmptyMerchantId"
    public static let EmptyCompanyLabel = "EmptyCompanyName"
    public static let InvalidCountryCode = "InvalidCountryCode"
    public static let InvalidCurrencyCode = "InvalidCurrencyCode"
    public static let LineItemTotalMismatch = "LineItemTotalMismatch"
    public static let DigitalWalletNotReady = "DigitalWalletNotReady"
    public static let NoAllowedPaymentNetworks = "NoAllowedCardNetworks"
    public static let GeneralError = "GeneralError"
}

File: src/pay_digitalwallets_sdk_android/android/src/main/kotlin/com/olo/flutter/digitalwallets/data/ErrorCodes.kt
// Copyright © 2025 Olo Inc. All rights reserved.
// This software is made available under the MIT License (See LICENSE file)
package com.olo.flutter.digitalwallets.data

class ErrorCodes {
    companion object {
        // Method call rejection code keys
        const val InvalidParameter = "InvalidParameter"
        const val MissingParameter = "MissingParameter"
        const val UnexpectedParameterType = "UnexpectedParameterType"
        const val SdkNotConfigured = "SdkNotConfigured"
        const val InvalidGooglePaySetup = "InvalidGooglePaySetup"
        const val DigitalWalletNotReady = "DigitalWalletNotReady"
        const val NoAllowedCardNetworks = "NoAllowedCardNetworks"
        const val UnexpectedError = "UnexpectedError"
        const val EmptyCompanyName = "EmptyCompanyName"
        const val InvalidCountryCode = "InvalidCountryCode"
        const val InvalidCurrencyCode = "InvalidCurrencyCode"
    }
}